### PR TITLE
implement rand and srand using the rand crate

### PR DESCRIPTION
**Pull request by sajattack**
The other rand functions specify specific algorithms or require a seedable, thread-safe rng, so they could not be implemented with the same approach.

### CODE REVIEW

File: src/stdlib/src/lib.rs [hunk 1]
```diff
@@ -1,15 +1,16 @@
-//! stdlib implementation for Redox, following http://pubs.opengroup.org/onlinepubs/7908799/xsh/stdlib.h.html
```
Comment (jackpot51) [inline src/stdlib/src/lib.rs]: Why was this comment line moved?

Comment (sajattack) [reply src/stdlib/src/lib.rs]: The compiler was complaining about it. It said no doc comments before the #![]s

Comment (jackpot51) [reply src/stdlib/src/lib.rs]: Why does the line have three slashes instead of two now?

Comment (sajattack) [reply src/stdlib/src/lib.rs]: I guess I did that by accident

Comment (jackpot51) [reply src/stdlib/src/lib.rs]: If you change it back to two can it move back to the top of the file?

Comment (sajattack) [reply src/stdlib/src/lib.rs]: Yes.

Comment (jackpot51) [reply src/stdlib/src/lib.rs]: :tada: Sweet! :+1:

**Pull request merged**